### PR TITLE
Fixes #724 : fixes publication export

### LINKED ISSUE #724: Regression: Export does not work anymore
Signaled by Microsoft on Windows S. Also experienced on MacOS. 

Steps to Reproduce:

1.      Launch the application.
2.     Select the three dots contextual menu on a book.
3.     Select “Export”.
4.     Notice no action is taken.

Expected result:

The controls within the application should work as expected and should not present “coming soon” messages.

### PR DESCRIPTION
**Pull request by Scarsniik**

### CODE REVIEW

**Comment by danielweck** [comment]
From the library/bookshelf view, in list mode (not the default cover image mode), when trying to export to a selected folder:

```
PublicationExportButton.tsx:65 Uncaught TypeError: this.props.onClick is not a function
    at PublicationExportButton.onExport (PublicationExportButton.tsx:65)
    at HTMLUnknownElement.callCallback (react-dom.development.js:347)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:397)
    at invokeGuardedCallback (react-dom.development.js:454)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:468)
    at executeDispatch (react-dom.development.js:600)
    at executeDispatchesInOrder (react-dom.development.js:622)
    at executeDispatchesAndRelease (react-dom.development.js:725)
    at executeDispatchesAndReleaseTopLevel (react-dom.development.js:733)
    at forEachAccumulated (react-dom.development.js:707)
PublicationExportButton.onExport @ PublicationExportButton.tsx:65
callCallback @ react-dom.development.js:347
invokeGuardedCallbackDev @ react-dom.development.js:397
invokeGuardedCallback @ react-dom.development.js:454
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:468
executeDispatch @ react-dom.development.js:600
executeDispatchesInOrder @ react-dom.development.js:622
executeDispatchesAndRelease @ react-dom.development.js:725
executeDispatchesAndReleaseTopLevel @ react-dom.development.js:733
forEachAccumulated @ react-dom.development.js:707
runEventsInBatch @ react-dom.development.js:750
runExtractedPluginEventsInBatch @ react-dom.development.js:881
handleTopLevel @ react-dom.development.js:5901
batchedEventUpdates$1 @ react-dom.development.js:21657
batchedEventUpdates @ react-dom.development.js:2348
dispatchEventForPluginEventSystem @ react-dom.development.js:5996
dispatchEvent @ react-dom.development.js:6025
2PublicationExportButton.tsx:76 Uncaught TypeError: this.props.onClick is not a function
    at PublicationExportButton.checkCancel (PublicationExportButton.tsx:76)
```

**Comment by danielweck** [comment]
Otherwise, mouse/keyboard interaction *do* display the file / folder selection dialog, but nothing happens (no exported publication).

**Comment by danielweck** [comment]
My two previous comments: same problems in the `develop` branch! (I am now checking before the recent refactor ...)

**Comment by danielweck** [comment]
Exact same problems in `ef1a83425e5ef58482893cb990a469e799d75348` (as in `develop`), so this is not specific to this PR.
https://github.com/readium/readium-desktop/commits/develop

So, the PR in this current state fixes the mouse interaction in the "cover image" library view, but we now need to fix the actual export logic ... @Scarsniik are you okay to continue working on this?

**Comment by Scarsniik** [comment]
Yes, i'll do this

File: src/main/services/catalog.ts [hunk 1]
```diff
@@ -241,7 +242,23 @@ export class CatalogService {
...
+            }
+        }
+
+        // Open a dialog to select a folder then copy the publication in it
+        let destinationPath: string = dialog.showOpenDialog(mainWindow, {
+            properties: ["openDirectory"],
+        })[0];
+
+        // If the selected path is a file then choose the directory containing this file
+        destinationPath = path.dirname(destinationPath);
```
Comment (danielweck) [inline src/main/services/catalog.ts]: This code fails to achieve the desired result. Tests:

`node -e "const path = require('path'); console.log(path.dirname('/folder1/folder2/file1.txt'));"`
=>
`/folder1/folder2`
(correct)


`node -e "const path = require('path'); console.log(path.dirname('/folder1/folder2'));"`
=>
`/folder1`
(incorrect, expected result is `/folder1/folder2`)

**Pull request merged**